src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java

```diff
@@ -212,24 +212,38 @@ private void validateDocument(DocumentLimitsConfig limits, ObjectNode doc, Strin
     }
 
     // Second: traverse to check for other constraints
-    validateObjectValue(limits, doc, 0);
+    validateObjectValue(limits, null, doc, 0);
   }
 
-  private void validateDocValue(DocumentLimitsConfig limits, JsonNode value, int depth) {
+  private void validateDocValue(
+      DocumentLimitsConfig limits, String referringPropertyName, JsonNode value, int depth) {
     if (value.isObject()) {
-      validateObjectValue(limits, value, depth);
+      validateObjectValue(limits, referringPropertyName, value, depth);
     } else if (value.isArray()) {
-      validateArrayValue(limits, value, depth);
+      validateArrayValue(limits, referringPropertyName, value, depth);
     } else if (value.isTextual()) {
       validateStringValue(limits, value);
     }
   }
 
-  private void validateArrayValue(DocumentLimitsConfig limits, JsonNode arrayValue, int depth) {
+  private void validateArrayValue(
+      DocumentLimitsConfig limits, String referringPropertyName, JsonNode arrayValue, int depth) {
     ++depth;
     validateDocDepth(limits, depth);
 
-    if (arrayValue.size() > limits.maxArrayLength()) {
+    // One special case: vector embeddings
+    if (DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD.equals(referringPropertyName)) {
```

Comment:
Should we first do if arrayValue.size() > limits.maxArrayLength() first, if true check the field name. This way, don't need to compare the field name for every array node.

Reply:
Was wondering which way to but yes that would probably slightly more efficient. Will change.